provenance/core.py

```diff
@@ -384,10 +415,16 @@ def file_load(id):
 
 def archive_file(filename, name=None, delete_original=False, custom_fields=None, preserve_ext=False):
     @provenance(archive_file=True, name=name or 'archive_file', preserve_file_ext=preserve_ext,
-                delete_original_file=delete_original, custom_fields=custom_fields)
-    def _archive_file(filename):
+                delete_original_file=delete_original, remove=['_archive_file_filename'],
+                custom_fields=custom_fields)
+
+    # we want artifacts created by archive_file to be invariant to the
+    # filename (see remove above) but # not the custom_fields.
+    # filename is still passed in so the hash of the file on disk can be
+    # computed as part of the id of the artifact.
+    def _archive_file(_archive_file_filename, custom_fields):
         return filename
```

Comment:
Okay, this [commit](https://github.com/Savvysherpa/provenance/commit/e154f2bafa1f7a8bb14643c1499a58e2b6543ef1) should fix that.